lisa/microsoft/testsuites/cpu/cpusuite.py

```diff
@@ -172,150 +173,300 @@ def verify_cpu_offline_network_workload(
             server.tools[Kill].by_name("iperf3", ignore_not_exist=True)
             client.tools[Kill].by_name("iperf3", ignore_not_exist=True)
 
+    # ---- CPUSuite helpers ----
+    def _clamp_channels(self, val: int) -> int:
+        return max(1, min(int(val), 64))
+
+    def _read_max_supported_structured(
+        self, node: Node, log: Optional[Logger]
+    ) -> Optional[int]:
+        try:
+            info = node.tools[Ethtool].get_device_channels_info("eth0", True)
+            candidates = [
+                getattr(info, "max_combined", None),
+                getattr(info, "max_channels", None),
+                getattr(info, "max_current", None),
+            ]
+
+            for c in candidates:
+                if isinstance(c, str) and c.isdigit():
+                    c = int(c)
+                if isinstance(c, int) and c > 1:
+                    val = self._clamp_channels(c)
+                    if log:
+                        log.debug(
+                            "Structured max channels found: %d (candidates=%s)",
+                            val,
+                            candidates,
+                        )
+                    return val
+            return None
+
+        except Exception as e:
+            if log:
+                log.debug("Structured channels info failed: %s", e)
+            return None
+
+    def _read_max_supported_raw(
+        self, node: Node, log: Optional[Logger]
+    ) -> Optional[int]:
+        try:
+            r = node.execute("ethtool -l eth0", sudo=True)
+            raw = r.stdout.strip()
+            if log:
+                log.debug("Raw `ethtool -l` output:\n%s", raw)
+
+            block = re.search(r"(?is)pre-?set\s+maximums:?(.*?)(?:current|$)", raw)
```

Comment:
1, add examples above on each pattern.
2. use named groups to get multiple values by one regexp.
3. use LISA internal lib to find patterns.